geomesa-native-api/src/main/java/org/locationtech/geomesa/api/GeoMesaQuery.java

```diff
@@ -0,0 +1,51 @@
+package org.locationtech.geomesa.api;
+
+import org.geotools.factory.CommonFactoryFinder;
+import org.opengis.filter.Filter;
+import org.opengis.filter.FilterFactory2;
+
+import java.util.Date;
+
+/**
+ * Represents a query to GeoMesa
+ */
+public class GeoMesaQuery {
+
+    private Filter filt = Filter.INCLUDE;
+
+    public static class GeoMesaQueryBuilder {
+        private static FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        private Double minx, miny, maxx, maxy;
+        private Date start;
+        private Date end;
+        public static GeoMesaQueryBuilder builder() {
```

Comment:
perhaps shorten to name the nested class Builder with a public no arg constructore to shorten life...though i've forgotten my java style.

```
GeomesaQuery query = new GeomesaQuery.Builder().within(1,2,3,4).during(foo).build()
```